### ISSUE TITLE
gogs  Repository Hooks tag event

### ISSUE THREAD

**Issue by fourcels** [drone/drone#2459]
Gogs  Repository Hooks tag event not have Reference Environment `DRONE_COMMIT_AUTHOR_EMAIL`

``` go
// drone/remote/gogs/helper.go 
// helper function that extracts the Build data from a Gogs push hook
func buildFromPush(hook *pushHook) *model.Build {
	avatar := expandAvatar(
		hook.Repo.URL,
		fixMalformedAvatar(hook.Sender.Avatar),
	)
	author := hook.Sender.Login
	if author == "" {
		author = hook.Sender.Username
	}
	sender := hook.Sender.Username
	if sender == "" {
		sender = hook.Sender.Login
	}

	return &model.Build{
		Event:     model.EventPush,
		Commit:    hook.After,
		Ref:       hook.Ref,
		Link:      hook.Compare,
		Branch:    strings.TrimPrefix(hook.Ref, "refs/heads/"),
		Message:   hook.Commits[0].Message,
		Avatar:    avatar,
		Author:    author,
		Email:     hook.Sender.Email,  // with DRONE_COMMIT_AUTHOR_EMAIL
		Timestamp: time.Now().UTC().Unix(),
		Sender:    sender,
	}
}

// helper function that extracts the Build data from a Gogs tag hook
func buildFromTag(hook *pushHook) *model.Build {
	avatar := expandAvatar(
		hook.Repo.URL,
		fixMalformedAvatar(hook.Sender.Avatar),
	)
	author := hook.Sender.Login
	if author == "" {
		author = hook.Sender.Username
	}
	sender := hook.Sender.Username
	if sender == "" {
		sender = hook.Sender.Login
	}

	return &model.Build{
		Event:     model.EventTag,
		Commit:    hook.After,
		Ref:       fmt.Sprintf("refs/tags/%s", hook.Ref),
		Link:      fmt.Sprintf("%s/src/%s", hook.Repo.URL, hook.Ref),
		Branch:    fmt.Sprintf("refs/tags/%s", hook.Ref),
		Message:   fmt.Sprintf("created tag %s", hook.Ref),
		Avatar:    avatar,
		Author:    author,
		Sender:    sender,
		Timestamp: time.Now().UTC().Unix(),
                 // without DRONE_COMMIT_AUTHOR_EMAIL
	}
}
```

**Comment by bradrydzewski** [comment]
this is resolved in the upcoming 0.9 alpha release:
https://github.com/drone/go-scm/blob/master/scm/driver/gogs/testdata/webhooks/tag_create.json.golden#L23

**Comment by bradrydzewski** [comment]
this is implemented, however, please beware that Gogs does not provide the commit message or the commit sha in the webhook, nor does it provide any way to fetch this information via the API. So this is supported, but will result in some missing data in the user-interface.
